operator/api/v1alpha2/keptnevaluationprovider_types.go

```diff
@@ -64,5 +64,5 @@ func init() {
 }
 
 func (p *KeptnEvaluationProvider) HasSecretKey() bool {
-	return p.Spec.SecretKeyRef == (corev1.SecretKeySelector{})
+	return p.Spec.SecretKeyRef != (corev1.SecretKeySelector{})
```

Comment:
I think we should in addition check if both the name of the secret, as well as the key within the secret are set, since both are required for the retrieval of the secret